src/compute/src/compute_state.rs

```diff
@@ -157,6 +173,10 @@ impl ComputeState {
         let command_history = ComputeCommandHistory::new(metrics.for_history(worker_id));
         let (hydration_tx, hydration_rx) = mpsc::channel();
 
+        // We always initialize as read_only=true. Only when we're explicitly
+        // allowed do we switch to doing writes.
+        let (read_only_tx, read_only_rx) = watch::channel(true);
```

Comment:
Can you add a dyncfg that allows initializing this to `false` too? This way we can derisk the rollout, if anything should be broken in the way how the `AllowWrites` command is propagated, we still have an escape hatch.

Alternatively, a dyncfg that makes persist_sink ignore read-only mode would work equally well.

Reply:
I can add it, but that adds more code that could be wrong:
- I don't think I can add a dyncfg at this point, because we didn't yet get any configuration updates at this point.
- If I want to do it in `persist_sink`, I first have to pipe through a new flag.

I'll quickly whip up the second option, and you can see if you like that.